modules/imgproc/include/opencv2/imgproc/hal/hal.hpp

```diff
@@ -60,6 +60,21 @@ struct  CV_EXPORTS MorphContext
 };
 
 
+CV_EXPORTS void resize(int src_type,
```

Comment:
dst_width and dst_height should be added to preserve OpenCV semantics. Because of floating-point arithmetics, sometimes we can not rely on HAL (or any library) to compute the size of destination image, we compute it ourselves and pass it explicitly.

1) if dst_width and dst_height are 0's, use inv_scale_x and inv_scale_y to compute it.
2) if dst_width and dst_height are non 0's, but inv_scale_x and inv_scale_y are 0's, compute them as ratios.
3) if all the parameters are non 0's, fill out the specified destination image - some parts of the source image might be cut off or replicated; the original OpenCV code does that. There can probably be a check that inv_scale_x and dst_width/src_width, inv_scale_y and dst_height/src_height are not very different.
4) in any other case report an error.